### ISSUE TITLE
Docs: Remove the word "reference" from subsections under "Reference Guides"

### ISSUE THREAD

**Issue by juanmaguitar** [WordPress/gutenberg#60782]
## What problem does this address?
To improve readability and make breadcrumbs shorter and more digestable

## What is your proposed solution?
To remove the word "reference" from subsections under "Reference Guides"

**Comment by ndiego** [comment]
Removing the word "reference" would also remove it from the page title. This might impact SEO and make page titles confusing. 

<img width="1426" alt="image" src="https://github.com/WordPress/gutenberg/assets/4832319/c9b5e513-d8e0-4eaa-ad87-1c869f5ea8eb">

**Comment by juanmaguitar** [comment]
@ndiego I agree with you that removing "reference" from the H1 could impact SEO and page titles.

To address both concerns (shorter breadcrumbs/sidebar while maintaining proper page titles), I've prepared a solution that adds support for a sidebar_label frontmatter field in the documentation generation process.

## How it works:

Add frontmatter to any markdown file to customize what appears in the sidebar/TOC independently from the page H1:

```
  ---
  sidebar_label: Interactivity API
  ---

  # Interactivity API Reference
```

In this example:
- The sidebar/breadcrumbs show: "Interactivity API" (clean and concise)
- The page title remains: "Interactivity API Reference" (better for SEO and clarity)

This allows us to remove redundant words like "reference" from subsections in the sidebar navigation, while keeping them in the actual page titles where they provide important context for users and search engines.

I've prepared a PR that implements this functionality: https://github.com/WordPress/gutenberg/pull/75068
